DPL: improve arrow::Dataset integration

ktf opened the pull request:
- Modularise filesystem to allow easier navigation and support for
  multiple formats.
- Add initial support to multiplex multiple tables on top of the same tree.
- Improve support for writing boolean fields.

alibuild commented:
Error while checking build/O2/fullCI for 475d25f53cd5f7bcc33bac9eadfa56118c403ebe at 2024-11-23 11:45:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/slc8_x86-64/GCC-Toolchain/v13.2.0-alice1-6/include/c++/13.2.0/bits/stl_construct.h:115:28: error: no matching function for call to 'construct_at(arrow::NumericArray<arrow::UInt8Type>*&)'
/sw/slc8_x86-64/GCC-Toolchain/v13.2.0-alice1-6/include/c++/13.2.0/bits/stl_construct.h:96:17: error: 'arrow::PrimitiveArray::PrimitiveArray()' is protected within this context
/sw/slc8_x86-64/GCC-Toolchain/v13.2.0-alice1-6/include/c++/13.2.0/bits/stl_construct.h:95:23: error: 'arrow::PrimitiveArray::PrimitiveArray()' is protected within this context
/sw/slc8_x86-64/GCC-Toolchain/v13.2.0-alice1-6/include/c++/13.2.0/bits/stl_construct.h:119:7: error: 'arrow::PrimitiveArray::PrimitiveArray()' is protected within this context
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13725/475d25f53cd5f7bcc33bac9eadfa56118c403ebe/build_O2_fullCI/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI for f37b9d7630ccc9fd357a5939bfb6700084b04514 at 2024-11-24 11:31:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/slc8_x86-64/GCC-Toolchain/v13.2.0-alice1-6/include/c++/13.2.0/bits/stl_construct.h:115:28: error: no matching function for call to 'construct_at(arrow::NumericArray<arrow::UInt8Type>*&)'
/sw/slc8_x86-64/GCC-Toolchain/v13.2.0-alice1-6/include/c++/13.2.0/bits/stl_construct.h:96:17: error: 'arrow::PrimitiveArray::PrimitiveArray()' is protected within this context
/sw/slc8_x86-64/GCC-Toolchain/v13.2.0-alice1-6/include/c++/13.2.0/bits/stl_construct.h:95:23: error: 'arrow::PrimitiveArray::PrimitiveArray()' is protected within this context
/sw/slc8_x86-64/GCC-Toolchain/v13.2.0-alice1-6/include/c++/13.2.0/bits/stl_construct.h:119:7: error: 'arrow::PrimitiveArray::PrimitiveArray()' is protected within this context
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13725/f37b9d7630ccc9fd357a5939bfb6700084b04514/build_O2_fullCI/pretty.html).

ktf committed:
DPL: improve arrow::Dataset integration

- Modularise filesystem to allow easier navigation and support for
  multiple formats.
- Add initial support to multiplex multiple tables on top of the same tree.
- Improve support for writing boolean fields.

ktf commented:
Errors unrelated. Only affects a working test. Merging.
